src/gc/impl/conservative/gc.d

```diff
@@ -1630,7 +1630,12 @@ struct Gcx
 
     void* smallAlloc(size_t size, ref size_t alloc_size, uint bits, const TypeInfo ti) nothrow
     {
-        immutable bin = binTable[size];
+        immutable trybin = binTable[size];
+        // ensure alignment of intended typeinfo can be satisfied by the bin size.
+        immutable bin = ti is null
+            ? trybin
+            : cast(immutable(byte))(binsize[trybin] % ti.talign == 0 ? trybin : trybin + 1);
```

Comment:
Looks ok, but as this only works for power of 2 alignments anyway, using `& (ti.align - 1)` should be more slightly efficient.

Reply:
Yes, true. All alignments must be positive power of 2 (statically enforced).